src/main/java/frc/robot/commands/IntakeSequence.java

```diff
@@ -1,46 +1,54 @@
 package frc.robot.commands;
 
-import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.ParallelCommandGroup;
+import edu.wpi.first.wpilibj2.command.ParallelRaceGroup;
+import edu.wpi.first.wpilibj2.command.RunCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.Cal;
 import frc.robot.subsystems.Intake;
 import frc.robot.subsystems.Lift;
+import frc.robot.subsystems.Lift.LiftPosition;
 
-public class IntakeSequence extends CommandBase {
+public class IntakeSequence extends SequentialCommandGroup {
   private Intake intake;
   private Lift lift;
 
-  public IntakeSequence(Intake intake, Lift lift) {
-    this.intake = intake;
-    this.lift = lift;
-  }
+  // deploy intake for specified amount of time
+  private ParallelRaceGroup deployIntake =
+      new RunCommand(intake::deploy, intake).withTimeout(Cal.Intake.AUTO_CLAMP_WAIT_TIME_SECONDS);
 
-  @Override
-  public void initialize() {
-    intake.deploy();
-    lift.setDesiredPosition(Lift.LiftPosition.GRAB_FROM_INTAKE);
-  }
+  // run intake infinitely (until parallel command ends)
+  private RunCommand runIntake = new RunCommand(intake::intakeGamePiece, intake);
+  
+  // trigger the lift to move to the intake position. This does not need a timeout because it is
+  // running in the parallel group, which is controlled by lift.readyToIntake()
+  private InstantCommand liftToIntakePos =
+      new InstantCommand(() -> lift.setDesiredPosition(LiftPosition.GRAB_FROM_INTAKE), lift);
 
-  @Override
-  public void execute() {
-    if (!lift.seeGamePiece()) {
-      intake.intakeGamePiece();
-    } else {
-      lift.closeGrabber();
-    }
-    // TODO fix this
-  }
+  // triggers the grabber to open when it is safe. Again, we do not care about time because it is in
+  // the parallel group.
+  private InstantCommand openGrabber = new InstantCommand(lift::openGrabber, lift);
 
-  @Override
-  public boolean isFinished() {
-    return lift.holdingGamePiece() ? true : false;
-  }
+  // run intake, move lift to intake position, and open the grabber, until the robot sees an object
+  // and the lift is in position
+  private ParallelRaceGroup intakePrep =
+      new ParallelCommandGroup(runIntake, liftToIntakePos, openGrabber).until(lift::readyToIntake);
```

Comment:
this should end when the lift is at the intaking position and when the intake sees a game piece, not when the lift sees a game piece. Also the until should be changed to something that makes sure that it will finish when both the ready to intake is good and the commands finish.

Reply:
Right now, all three commands are instant commands (I changed runIntake to an instant command); should I add end conditions to them?